# Code Repo for Cell Village Project for Xu  
Wilson wrote it on 10/29/2022  
  
### List of scripts used in the following orders:  
#### 1. scRNAseq.sh  
Purpose: To run cellranger analysis on the FASTQ files. Only for one run.
How to run:  
```
sh scRNAseq.sh <samplename>  
  
Details:
<samplename> : The name of the sample used to identify the specific pairs of fastq files. See FASTQ File naming.

Within the shell script:
REF: reference genome built for cellrange (hg38).
FQ: the full path to the fastq file folder. Therefore, $S1 refers to the folder containing the fastq file.
```  
  
Output:
A folder with complete cellranger run.  

Note:
Possible to rerun in case the analysis was cutoff. cellranger is able to continue the run from where it stops.  

#### 2. demuxlet.sh  
Purpose: To run demultiplexing of cells to donor origin using genotype.  
How to run:
```
sh demuxlet.sh  

Details:
No user argument required, but changing of parameters within the script might be required.  

Within the shalle script:
BAM: the bam file in question (suggesting using position-sorted bam file generated by cellranger)  
VCF: topmed vcf files (suggest that you use the entire VCF from Topmed
OUTPUT: the output prefix 

Note:
Add demuxlet to the global $PATH  
 
```  
  
Output:
```  
demuxlet.best: (for downstream analysis). The BEST attribute should refer to the best assigned donor id in VCF file 
other text files (for references)
```  

Note:  
I don't find reducing the variants have significant impact on speed. You can try with exome / just epressed genes. However, reducing the number of individuals in the vcf files help to reduce running speed.  
  
  
#### 3. seurat.R  
Purpose: Generate all kinds of single-cell analysis for the EB datasets using R seurat.  
How to run:  
```
Rscript seurat.R
! Please do this interactively especially to determine the QC cut-off! 
! Run in each folder only !

Input: outs/filtered_feature_bc_matrix (sometimes I use raw, but it shouldnt matter because I will filter aagain)  

Within the seurat script:  
Look within the script   (bin/seurat.R)

Output:  
heart.rds  (use for integration)  

```  
  



### List of R object and its meaning:  

